### ISSUE TITLE
Code command and environment not working with SSH on windows but fine with mac

### ISSUE THREAD

**Issue by Ali-Hudson** [microsoft/vscode#142628]
I am working remotely on an HPC using SSH. 

I have a Windows machine at home and a Mac at my workplace, both of which connect remotely to the high performance computing using the same code successfully (ssh user@login followed by a password). 

When connected, VS Code works as well on the Mac as if I were just looking at local files. The explorer shows the directories I am working in and I can easily open a file by clicking it in the explorer, or using code filename in the terminal.

However, from the Windows machine, I can connect to the HPC, but the explorer does not load the directories I am working in, and when running code filename, I get "-bash: code: command not found" instead. 
I can still navigate directories and execute commands but I cannot view a file except to open it in Nano or another text editor.

If I'm not connected to SSH, I can use code to open a local file without issue. 

I have done a clean install on the windows machine, but nothing seems to help. I don't understand why it is fine from the Mac and not the Windows when it works locally. 

Can anyone suggest a solution or explain why this isn't working?

Steps to reproduce: 
1. open VS code terminal
2. Connect remotely to server using ssh user@login, and enter the password
3. Navigate to any file with permission to open it.
4. In terminal: code filename

**Comment by shern2** [comment]
I experienced the same: Unable to use the `code` command when SSH-ing from Windows machine into any of my 4 Ubuntu machines.

**Comment by chrmarti** [comment]
This is only expected to work when you use the Remote-SSH extension to connect to the SSH server. Are you using that?

**Comment by shern2** [comment]
Yup. Via the MS remore SSH extension.

**Comment by Ali-Hudson** [comment]
Thanks chrmarti! I thought I had settings sync on and would have whatever was on my mac on my windows pc but I didn't!

For anyone else having the issue, the solution was:

1. Go to Extensions (ctrl+shift+f)
2. Search for Remote-SSH and install it
3. In the command pallete, select Remote-SSH Add New SSH Host
4. Type in your usual command (ssh user@login)
5. Select an OS for your host. I just assumed the HPC I use was Linux which seems fine. 
6. Go back to the command pallete and select Remote-SSH Connect to Host. 
7. Select the host you just added and it should prompt you for any passwords and then bring you to your home directory.
8. This automatically fixed the <code filename> not working issue, but I did have to manually tell it to load the directories into the explorer...
9. With the explorer bar open (or the Get Started tab), click Open Folder. It will bring up the path to the folder you are in, so you may wish to navigate to the root directory of your workspace before doing this. Click OK. For me this opened a new window and I had to put my password in again. After this everything was fine! :)

**Comment by shern2** [comment]
Hi, for me, I've been happily using the `remote SSH` extension "ms-vscode-remote.remote-ssh" for >1 year now.
I've not had problems before. This is first time I have encountered this.

**Success case (Ubuntu host1)**: I've just tested SSH from Ubuntu host1 to Ubuntu host2. After SSH was successful, am able to run `code [dir]` which opens a new workspace with that directory `[dir]` or `code [file]` opens the file `[file]`.
 
**Failure case (Windows host)**: However, when I SSH from my Windows host to Ubuntu host2, and ran `code [dir]`, nothing happens. When I check `PATH`, while I see it was pointing to the vscode server bin `/home/[myUsername]/.vscode-server/bin/d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03/bin`, but it only has the following
```
code-server  helpers  remote-cli
```
and does not point to any `code` executable.
NOTE: prior to this,  I have done: 
Full Windows Update -> restart laptop -> ran both `Kill VS Code Server on Host` as well as `Kill Local Connection Server For Host` -> `connect to SSH` 

So, I'm at my wits end..


Both my Windows host and Ubuntu host1 has the Remote SSH extension
`ms-vscode-remote.remote-ssh` v0.70.0.

My Windows VSCode details:
```
Version: 1.64.1 (user setup)
Commit: d6ee99e4c045a6716e5c653d7da8e9ae6f5a8b03
Date: 2022-02-07T17:16:22.147Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.19044
```

My Ubuntu VSCode details:
```
Version: 1.63.2
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:39:46.686Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Linux x64 5.4.0-97-generic
```

Would appreciate some help on this.
Thank you!
